### ISSUE TITLE
GetLink + ChoiceLink how to use?

### ISSUE THREAD

**Issue by AmeBel** [opencog/atomspace#326]
My Intention was to use ChoiceLink to get either of the EvaluationLinks, if they are present in the atomspace. also tried
* `(ChoiceLink (PresentLink (EvaluationLink ...)))`
* `(OrLink  (PresentLink (EvaluationLink ...))) ; just in case, was expecting an error `

both also gave empty `SetLink`

#### Expected result
```scheme
(SetLink
    (Node "OpenPsi: Energy-rule-xej90")
    (Node "OpenPsi: Energy-rule-pG1ez")
)
```

#### Content of the atomspace
```scheme
 (EvaluationLink
   (PredicateNode "OpenPsi:Decrease")
   (ListLink
      (Node "OpenPsi: Energy-rule-xej90")
      (ConceptNode "OpenPsi:Energy")
   )
)
(EvaluationLink
   (PredicateNode "OpenPsi:Increase")
   (ListLink
      (Node "OpenPsi: Energy-rule-pG1ez")
      (ConceptNode "OpenPsi:Energy")
   )
)
```
#### Commands run
```scheme
guile> (define get-nodes1
    (GetLink
       (TypedVariableLink
          (VariableNode "x")
          (TypeNode "Node")
       )
       (ChoiceLink
          (EvaluationLink
             (PredicateNode "OpenPsi:Increase")
             (ListLink
                (VariableNode "x")
                (ConceptNode "OpenPsi:Energy")
             )
          )
          (EvaluationLink
             (PredicateNode "OpenPsi:Decrease")
             (ListLink
                (VariableNode "x")
                (ConceptNode "OpenPsi:Energy")
             )
          )
       )
    )
)

guile> (cog-execute! get-nodes1)                                   
(SetLink                                                          
)                                                                 
```

**Comment by linas** [comment]
Strange; that should work. It works if you skip the variable declaration.

**Comment by linas** [comment]
Yeah, I see the problem.  Working on it.

**Comment by linas** [comment]
Fixed in pul req #327

**Comment by AmeBel** [comment]
thanks for the fix
